image/image.go

```diff
@@ -166,15 +231,54 @@ func createRuntimeBundle(w walker, dest, refName, rootfs string) error {
 		return err
 	}
 
-	m, err := findManifest(w, ref)
-	if err != nil {
-		return err
+	if ref.MediaType == validRefMediaTypes[0] {
+		m, err := findManifest(w, ref)
+		if err != nil {
+			return err
+		}
+
+		if err := m.validate(w); err != nil {
+			return err
+		}
+
+		return createRuntimebundle(w, m, dest, rootfs)
 	}
 
-	if err = m.validate(w); err != nil {
-		return err
+	if ref.MediaType == validRefMediaTypes[1] {
+		ml, err := findManifestList(w, ref)
+		if err != nil {
+			return err
+		}
+
+		if err := ml.validate(w); err != nil {
+			return nil
+		}
+
+		if len(ml.Manifests) == 0 {
+			fmt.Println("warning: no manifests found")
+			return nil
+		}
+
+		for _, manifest := range ml.Manifests {
+			m, err := findManifest(w, &(manifest.Descriptor))
+			if err != nil {
+				return err
+			}
+
+			if err := m.validate(w); err != nil {
+				return err
+			}
+
+			if err := createRuntimebundle(w, m, dest, rootfs); err != nil {
+				return err
```

Comment:
On creating runtime bundle, I don't think we should do this directly, because the manifest list contains different manifest of architecture or platform, we should not unpack all the manifest to a same `dest`. I think we should add a flag to `oci-create-runtime-bundle` and `oci-unpack`  to specify the arch we want to unpack or the `oci-create-runtime-bundle` and `oci-unpack` detect the target arch and unpack the target arch manifest.
ping @opencontainers/image-tools-maintainers